doc/source/reference/c-api/array.rst

```diff
@@ -2884,11 +3172,279 @@ Data-type descriptors
+   instance.
+
+.. c:macro:: NPY_DT_NUMERIC
+
+   Indicates the DType is represents a numerical value.
+
+
+.. _dtype-slots:
+
+Slot IDs and API Function Typedefs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These IDs correspond to slots in the DType API and are used to identify
+implementations of each slot from the items of the ``slots`` array
+member of ``PyArrayDTypeMeta_Spec`` struct.
+
+.. c:macro:: NPY_DT_discover_descr_from_pyobject
+
+.. c:type:: PyArray_Descr *(PyArrayDTypeMeta_DiscoverDescrFromPyobject)( \
+                PyArray_DTypeMeta *cls, PyObject *obj)
+
+   Used during DType inference to find the correct DType for a given
+   PyObject. Must return a descriptor instance appropriate to store the
+   data in the python object that is passed in. *obj* is the python object
+   to inspect and *cls* is the DType class to create a descriptor for.
+
+.. c:macro:: NPY_DT_default_descr
+
+.. c:type:: PyArray_Descr *(PyArrayDTypeMeta_DefaultDescriptor)( \
+                PyArray_DTypeMeta *cls)
+
+   Returns the default descriptor instance for the DType. Must be
+   defined for parametric data types. Non-parametric data types return
+   the singleton by default.
+
+.. c:macro:: NPY_DT_common_dtype
+
+.. c:type:: PyArray_DTypeMeta *(PyArrayDTypeMeta_CommonDType)( \
+                PyArray_DTypeMeta *dtype1, PyArray_DTypeMeta *dtype2)
+
+   Given two input DTypes, determines the appropriate "common" DType
+   that can store values for both types. Returns ``Py_NotImplemented``
+   if no such type exists.
+
+.. c:macro:: NPY_DT_common_instance
+
+.. c:type:: PyArray_Descr *(PyArrayDTypeMeta_CommonInstance)( \
+               PyArray_Descr *dtype1, PyArray_Descr *dtype2)
+
+   Given two input descriptors, determines the appropriate "common"
+   descriptor that can store values for both instances. Returns ``NULL``
+   on error.
+
+.. c:macro:: NPY_DT_ensure_canonical
+
+.. c:type:: PyArray_Descr *(PyArrayDTypeMeta_EnsureCanonical)( \
+                PyArray_Descr *dtype)
+
+   Returns the "canonical" representation for a descriptor instance. The
+   notion of a canonical descriptor generalizes the concept of byte
+   order, in that a canonical descriptor always has native byte
+   order. If the descriptor is already canonical, this function returns
+   a new reference to the input descriptor.
+
+.. c:macro:: NPY_DT_setitem
+
+.. c:type:: int(PyArrayDTypeMeta_SetItem)(PyArray_Descr *, PyObject *, char *)
+
+   Implements scalar setitem for an array element given a PyObject.
+
+.. c:macro:: NPY_DT_getitem
+
+.. c:type:: PyObject *(PyArrayDTypeMeta_GetItem)(PyArray_Descr *, char *)
+
+   Implements scalar getitem for an array element. Must return a python
+   scalar.
+
+.. c:macro:: NPY_DT_get_clear_loop
+
+   If defined, sets a traversal loop that clears data in the array. This
+   is most useful for arrays of references that must clean up array
+   entries before the array is garbage collected. Implements
+   ``PyArrayMethod_GetLoop``.
```

Comment:
I'm confused why this implements `PyArrayMethod_GetLoop` when above these were discussed under traverse loops.

Reply:
thanks, this is a typo.